File: backend/src/AccountOS.Domain/Entities/Customer.cs
using AccountOS.Domain.Common;
using AccountOS.Domain.Enums;

namespace AccountOS.Domain.Entities;

/// <summary>
/// Cari hesap (Müşteri/Tedarikçi) entity
/// </summary>
public class Customer : TenantEntity
{
    /// <summary>Cari hesap kodu (otomatik: C-0001)</summary>
    public string Code { get; set; } = string.Empty;
    
    /// <summary>Cari hesap adı/unvanı</summary>
    public string Name { get; set; } = string.Empty;
    
    /// <summary>Cari hesap tipi (Bireysel/Kurumsal)</summary>
    public CustomerType Type { get; set; }
    
    // İletişim Bilgileri
    /// <summary>E-posta adresi</summary>
    public string? Email { get; set; }
    
    /// <summary>Telefon</summary>
    public string? Phone { get; set; }
    
    /// <summary>Cep telefonu</summary>
    public string? MobilePhone { get; set; }
    
    /// <summary>Web sitesi</summary>
    public string? Website { get; set; }
    
    // Vergi Bilgileri
    /// <summary>Vergi numarası (kurumsal için)</summary>
    public string? TaxNumber { get; set; }
    
    /// <summary>Vergi dairesi</summary>
    public string? TaxOffice { get; set; }
    
    /// <summary>TC Kimlik No (bireysel için)</summary>
    public string? IdentityNumber { get; set; }
    
    // Adres Bilgileri
    /// <summary>Fatura adresi</summary>
    public string? BillingAddress { get; set; }
    
    /// <summary>Sevkiyat adresi</summary>
    public string? ShippingAddress { get; set; }
    
    /// <summary>Şehir</summary>
    public string? City { get; set; }
    
    /// <summary>Ülke</summary>
    public string? Country { get; set; }
    
    /// <summary>Posta kodu</summary>
    public string? PostalCode { get; set; }
    
    // Finansal Bilgiler
    /// <summary>Varsayılan para birimi</summary>
    public string Currency { get; set; } = "TRY";
    
    /// <summary>Kredi limiti</summary>
    public decimal CreditLimit { get; set; }
    
    /// <summary>Vade gün sayısı (örn: 30 gün)</summary>
    public int PaymentTermDays { get; set; }
    
    /// <summary>Güncel bakiye (Alacak: pozitif, Borç: negatif)</summary>
    public decimal CurrentBalance { get; set; }
    
    // Durum
    /// <summary>Aktif mi?</summary>
    public bool IsActive { get; set; } = true;
    
    /// <summary>Bloke edilmiş mi?</summary>
    public bool IsBlocked { get; set; }
    
    /// <summary>Bloke nedeni</summary>
    public string? BlockReason { get; set; }
    
    // Notlar
    /// <summary>Genel notlar</summary>
    public string? Notes { get; set; }
}

File: frontend/src/types/customer.ts
// Müşteri tipleri
export interface Customer {
  id: string;
  companyId: string;
  code: string; // Müşteri kodu (AUTO: C-0001)
  name: string;
  type: CustomerType;
  
  // Contact
  email?: string;
  phone?: string;
  mobilePhone?: string;
  website?: string;
  
  // Tax Info
  taxNumber?: string;
  taxOffice?: string;
  identityNumber?: string; // TC Kimlik (bireysel için)
  
  // Address
  billingAddress?: string;
  shippingAddress?: string;
  city?: string;
  country?: string;
  postalCode?: string;
  
  // Financial
  currency: string;
  creditLimit: number;
  paymentTermDays: number; // Vade gün sayısı
  currentBalance: number; // Güncel bakiye (Alacak/Borç)
  
  // Status
  isActive: boolean;
  isBlocked: boolean;
  blockReason?: string;
  notes?: string;
  
  // Audit
  createdAt: string;
  updatedAt?: string;
}

export enum CustomerType {
  Individual = 0, // Bireysel
  Corporate = 1,  // Kurumsal
}

// Create/Update için
export interface CreateCustomerRequest {
  name: string;
  type: CustomerType;
  email?: string;
  phone?: string;
  mobilePhone?: string;
  taxNumber?: string;
  taxOffice?: string;
  identityNumber?: string;
  billingAddress?: string;
  shippingAddress?: string;
  city?: string;
  country?: string;
  currency: string;
  creditLimit?: number;
  paymentTermDays?: number;
  notes?: string;
}

export type UpdateCustomerRequest = CreateCustomerRequest;

// Customer detail response
export interface CustomerDetail extends Customer {
  // Balance history
  balanceHistory: BalanceHistoryItem[];
  
  // Recent invoices
  recentInvoices: Invoice[];
  
  // Recent payments
  recentPayments: Payment[];
  
  // Addresses
  addresses: CustomerAddress[];
  
  // Statistics
  statistics: CustomerStatistics;
}

export interface BalanceHistoryItem {
  date: string;
  description: string;
  debit: number;    // Borç
  credit: number;   // Alacak
  balance: number;  // Bakiye
  referenceType: 'Invoice' | 'Payment';
  referenceId: string;
  referenceNumber: string;
}

export interface CustomerAddress {
  id: string;
  type: 'Billing' | 'Shipping';
  title: string;
  address: string;
  city: string;
  district?: string;
  postalCode?: string;
  country: string;
  isDefault: boolean;
}

export interface CustomerStatistics {
  totalInvoices: number;
  totalInvoiceAmount: number;
  totalPayments: number;
  totalPaymentAmount: number;
  averageInvoiceAmount: number;
  averagePaymentDays: number;
  lastInvoiceDate?: string;
  lastPaymentDate?: string;
}

export interface Invoice {
  id: string;
  invoiceNumber: string;
  invoiceDate: string;
  type: 'Sales' | 'Purchase';
  isOfficial: boolean;
  grandTotal: number;
  currency: string;
}

export interface Payment {
  id: string;
  paymentNumber: string;
  paymentDate: string;
  type: 'Receipt' | 'Payment';
  method: string;
  amount: number;
  currency: string;
}

